src/Hl7.Fhir.ElementModel/SourceNode.cs

```diff
@@ -15,149 +15,106 @@
+        /// <param name="recursive"></param>
+        /// <param name="annotationsToCopy">Maybe: if null - copy all, if empty, copy none, else specifcy which</param>
+        /// <returns></returns>
+        public static SourceNode FromNode(ISourceNode node, bool recursive = true, IEnumerable<Type> annotationsToCopy = null)
+            => buildNode(node, recursive, annotationsToCopy);
+
+        private static SourceNode buildNode(ISourceNode node, bool recursive, IEnumerable<Type> annotationsToCopy)
         {
             var me = new SourceNode(node.Name, node.Text);
-            me.AddRange(node.Children().Select(c => buildNode(c)));
-            return me;
-        }
 
+            var rts = node.Annotation<IResourceTypeSupplier>();
+            if (rts != null)
+                me.ResourceType = rts.ResourceType;
 
-        public ISourceNode Clone()
-        {
-            var copy = new SourceNode(Name, Text, _children, ResourceType);
+            foreach (var t in annotationsToCopy ?? Enumerable.Empty<Type>())
+                foreach (var ann in node.Annotations(t))
+                    me.AddAnnotation(ann);
 
-            if (_annotations.IsValueCreated)
-                copy.annotations.AddRange(annotations);
+            if (recursive)
+                me.AddRange(node.Children().Select(c => buildNode(c, recursive: true, annotationsToCopy: annotationsToCopy)));
 
-            return copy;
+            return me;
         }
 
-        private Lazy<List<object>> _annotations = new Lazy<List<object>>(() => new List<object>());
-        private List<object> annotations { get { return _annotations.Value; } }
+        public SourceNode Clone()
+        {
+            var copy = new SourceNode(Name, Text, ResourceType)
+            {
+                Parent = Parent
+            };
 
-        public IElementDefinitionSummary Definition => throw new NotImplementedException();
+            copy.AddRange(Children().Cast<SourceNode>().Select(c => c.Clone()));
 
-        public ChildNodes this[string name] => new ChildNodes(_children.Where(c=>c.Name == name).ToList());
+            if (HasAnnotations)
+                copy.AnnotationsInternal.AddRange(AnnotationsInternal);
 
-        public SourceNode this[int index] => _children[index];
+            return copy;
+        }
 
         public IEnumerable<object> Annotations(Type type)
         {
             return type == typeof(SourceNode) || type == typeof(ISourceNode) || type == typeof(IResourceTypeSupplier)
                 ? (new[] { this })
-                : annotations.OfType(type);
-        }
-
-        public void AddAnnotation(object annotation)
-        {
-            annotations.Add(annotation);
-        }
-
-        public void RemoveAnnotations(Type type)
-        {
-            annotations.RemoveOfType(type);
+                : AnnotationsInternal.OfType(type);
         }
-    }
 
-
-    public class ChildNodes : IEnumerable<ISourceNode>
```

Comment:
ChildNodes will be removed? Is that not breaking the interface?

Reply:
I guess I have. Unfortunately, I have no way to keep this class.  Well, technically we could paste it back in without using it ourselves and then marking it [Obsolete] I guess?